qiskit/circuit/quantumcircuit.py

```diff
@@ -1150,10 +1203,31 @@ def __init__(
         self._duration = None
         self._unit = "dt"
         self.metadata = {} if metadata is None else metadata
-        """Arbitrary user-defined metadata for the circuit.
+        """Arbitrary user-defined dictionary of metadata for the circuit.
 
         Qiskit will not examine the content of this mapping, but it will pass it through the
-        transpiler and reattach it to the output, so you can track your own metadata."""
+        transpiler and reattach it to the output, so you can track your own metadata.
+        
+        Example:
+
+            .. plot::
+                :include-source:
+                :nofigs:
+
+                from qiskit import QuantumRegister, ClassicalRegister, QuantumCircuit
+
+                q = QuantumRegister(2)
+                c = ClassicalRegister(2)
+                qc = QuantumCircuit(q, c)
+
+                qc.metadata = {'experiment_type': 'Bell state experiment'}
+
+                print(qc.metadata)
```

Comment:
Same here as with `.name`. Also, do you know if this is needed since you've also added an example in [metadata property](https://github.com/Qiskit/qiskit/pull/13300/files#diff-f8bf228c13afac54d0487df4391520ab5b0efbe5649e447475d99dd04ce06c89R1609)? I'm not sure how it works